src/Symfony/Bundle/FrameworkBundle/Command/RouterDebugCommand.php

```diff
@@ -172,4 +178,59 @@ private function getAvailableFormatOptions(): array
     {
         return (new DescriptorHelper())->getFormats();
     }
+
+    private function sortBy(RouteCollection $routeCollection, string $column): RouteCollection
+    {
+        $routesArray = $routeCollection->all();
+        $sortedRoutes = new RouteCollection();
+
+        if ('name' === $column) {
+            \ksort($routesArray);
+            foreach ($routesArray as $name => $route) {
+                $sortedRoutes->add($name, $route);
+                $sortedRoutes->addAlias($route->getDefault('_controller'), $name);
+            }
+
+            return $sortedRoutes;
+        }
+
+        $helperArray = [];
+
+        if ('method' === $column) {
+            foreach ($routesArray as $name => $route) {
+                $methods = empty($route->getMethods()) ? 'ANY' : \implode('|', $route->getMethods());
+                $helperArray[$methods][$name] = $route;
+            }
+        }
+
+        if ('scheme' === $column) {
+            foreach ($routesArray as $name => $route) {
+                $scheme = empty($route->getSchemes()) ? 'ANY' : \implode('|', $route->getSchemes());
+                $helperArray[$scheme][$name] = $route;
+            }
+        }
+
+        if ('host' === $column) {
+            foreach ($routesArray as $name => $route) {
+                $helperArray[$route->getHost()][$name] = $route;
+
+            }
+        }
+
+        if ('path' === $column) {
+            foreach ($routesArray as $name => $route) {
+                $helperArray[$route->getPath()][$name] = $route;
+            }
+        }
+
+        \ksort($helperArray);
```

Comment:
Instead of creating an array and using `ksort`, you can use `usort` with a callback that   compare the sort key of routes 2-by-2. It think there would be less array variables involved.